extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,270 @@
+        if flights_exist:
+            await delete_collection(flights_ref)
+
+        # initialize collections
+        for airport in airports:
+            await self.__client.collection("airports").document(str(airport.id)).set(
+                {
+                    "iata": airport.iata,
+                    "name": airport.name,
+                    "city": airport.city,
+                    "country": airport.country,
+                }
+            )
+        for amenity in amenities:
+            await self.__client.collection("amenities").document(str(amenity.id)).set(
+                {
+                    "name": amenity.name,
+                    "description": amenity.description,
+                    "location": amenity.location,
+                    "terminal": amenity.terminal,
+                    "category": amenity.category,
+                    "hour": amenity.hour,
+                    "content": amenity.content,
+                    "embedding": amenity.embedding,
+                }
+            )
+        for flight in flights:
+            await self.__client.collection("flights").document(str(flight.id)).set(
+                {
+                    "airline": flight.airline,
+                    "flight_number": flight.flight_number,
+                    "departure_airport": flight.departure_airport,
+                    "arrival_airport": flight.arrival_airport,
+                    "departure_time": flight.departure_time,
+                    "arrival_time": flight.arrival_time,
+                    "departure_gate": flight.departure_gate,
+                    "arrival_gate": flight.arrival_gate,
+                }
+            )
+
+    async def export_data(
+        self,
+    ) -> tuple[list[models.Airport], list[models.Amenity], list[models.Flight]]:
+        airport_docs = self.__client.collection("airports").stream()
+        amenities_docs = self.__client.collection("amenities").stream()
+        flights_docs = self.__client.collection("flights").stream()
+
+        airports, amenities, flights = [], [], []
+
+        for doc in airport_docs:
+            airport_dict = doc.to_dict()
+            airport_dict["id"] = doc.id
+            airports.append(models.Airport.model_validate(airport_dict))
+
+        for doc in amenities_docs:
+            amenity_dict = doc.to_dict()
+            amenity_dict["id"] = doc.id
+            amenities.append(models.Amenity.model_validate(amenity_dict))
+
+        for doc in flights_docs:
+            flight_dict = doc.to_dict()
+            flight_dict["id"] = doc.id
+            flights.append(models.Flight.model_validate(flight_dict))
+
+        return airports, amenities, flights
+
+    async def get_airport_by_id(self, id: int) -> Optional[models.Airport]:
+        query = self.__client.collection("airports").where(
+            filter=FieldFilter("id", "==", id)
+        )
+        return models.Airport.model_validate(query.stream().to_dict())
```

Comment:
Does query need to be awaited somehow?

Reply:
added await

Reply:
Ok, I keep seeing this and I don't understand it. 

Shouldn't we need to await the stream or call `.get` and await that? This example shows them awaiting from `.get()`: 
https://cloud.google.com/firestore/docs/query-data/get-data

This example shows them awaiting `.stream()`: https://cloud.google.com/firestore/docs/query-data/get-data#get_all_documents_in_a_collection

Reply:
I don't see this example awating `.stream()`? It uses `async for` to loop the stream but does not await? When I tried to add await before `.stream()`, it gave me error `TypeError: object async_generator can't be used in 'await' expression`

Reply:
I think `.get()` needs to be awaited because it returns the document snapshots themselves, while `.stream()` only returns the generator object to be consumed